src/txn/txn_rollback_to_stable.c

```diff
@@ -145,6 +148,43 @@ __rollback_row_add_update(WT_SESSION_IMPL *session, WT_PAGE *page, WT_ROW *rip,
     return (ret);
 }
 
+/*
+ * __rollback_check_if_txnid_non_committed --
+ *     Check if the transaction id is non committed.
+ */
+static bool
+__rollback_check_if_txnid_non_committed(WT_SESSION_IMPL *session, uint64_t txnid)
+{
+    WT_CONNECTION_IMPL *conn;
+    bool found;
+
+    conn = S2C(session);
+
+    /*
+     * Return false when the recovery snapshot count is 0, which means there is no uncommitted
+     * transaction ids.
+     */
+    if (conn->recovery_ckpt_snapshot_count == 0)
+        return false;
+
+    WT_BINARY_SEARCH(
+      txnid, conn->recovery_ckpt_snapshot, conn->recovery_ckpt_snapshot_count, found);
+
+    if (!found) {
+        /*
+         * Snapshot data:
+         *	ids < recovery_ckpt_snap_min are committed,
+         *	ids > recovery_ckpt_snap_max are non committed,
+         *	everything else is committed unless it is in the recovery_ckpt_snapshot array.
+         */
+        if (txnid < conn->recovery_ckpt_snap_min)
```

Comment:
Comparing against `snap_min` and `snap_max` are simple checks, better to perform them first than after a binary search. In normal visibility check also it will be performed similarly as said.